### PR TITLE
Makefile: fix the generation of `redact_safe.md`

### LINKED ISSUE #66401: docs: spurious diffs for redact_safe.md
**Describe the problem**

I'm seeing spurious diffs for `redact_safe.md`.
```
$ rm docs/generated/redact_safe.md; make docs/generated/redact_safe.md
$ git diff
diff --git i/docs/generated/redact_safe.md w/docs/generated/redact_safe.md
index ea1d855d26..05e7dce774 100644
--- i/docs/generated/redact_safe.md
+++ w/docs/generated/redact_safe.md
@@ -34,6 +34,8 @@ pkg/storage/enginepb/mvcc3.go | `MVCCStatsDelta`
 pkg/storage/enginepb/mvcc3.go | `*MVCCStats`
 pkg/util/hlc/timestamp.go | `Timestamp`
 pkg/util/hlc/timestamp.go | `ClockTimestamp`
+vendor/github.com/cockroachdb/pebble/internal/humanize/humanize.go | `FormattedString`
+docs/generated/redact_safe.md:57:vendor/github.com/cockroachdb/pebble/metrics.go:324:  // have been registered as safe with redact.RegisterSafeType and does not
 pkg/util/log/redact.go | `reflect.TypeOf(true)`
 pkg/util/log/redact.go | `reflect.TypeOf(123)`
 pkg/util/log/redact.go | `reflect.TypeOf(int8(0))`
@@ -53,3 +55,4 @@ pkg/util/log/redact.go | `reflect.TypeOf(time.Time{})`
 pkg/util/log/redact.go | `reflect.TypeOf(time.Duration(0))`
 pkg/util/log/redact.go | `reflect.TypeOf(encodingtype.T(0))`
 pkg/util/log/redact.go | `reflect.TypeOf(Channel(0))`
+vendor/github.com/cockroachdb/pebble/metrics.go:324:   // have been registered as safe with redact.RegisterSafeType and does not
```

Jira issue: CRDB-8026

### PR DESCRIPTION
**Pull request by knz**
Fixes #66401. 
cc @cockroachdb/dev-inf 

For context / reminder, the file `redact_safe.md` exists as sentinel
so that any changes in the list of types considered "safe for
reporting" triggers a diff in that file and thus a PR review
notification to the appropriate team(s).

Prior to this patch, the file was generated only for the files inside
the crdb repository. Since then, vendored packages (e.g. pebble) have
started using the redact package and registering their own safe
types. We want to see these reported in the sentinel files as well.
This patch achieves this.

Additionally, the output was somewhat non-deterministic, as there is
no guarantee that the output of `git grep` is sorted. This patch fixes
that by adding an explicit `sort` step.

Finally, the patch ensures that mentions of `redact.RegisterSafeType`
inside comment-only lines are ignored.

Release note: None

### CODE REVIEW

File: docs/generated/redact_safe.md [hunk 1]
```diff
@@ -10,46 +10,47 @@ pkg/kv/kvserver/raft.go | `SnapshotRequest_Type`
...
 pkg/sql/sem/tree/table_ref.go | `ColumnID`
-pkg/storage/enginepb/mvcc3.go | `MVCCStatsDelta`
+pkg/sql/sem/tree/table_ref.go | `ID`
 pkg/storage/enginepb/mvcc3.go | `*MVCCStats`
-pkg/util/hlc/timestamp.go | `Timestamp`
+pkg/storage/enginepb/mvcc3.go | `MVCCStatsDelta`
 pkg/util/hlc/timestamp.go | `ClockTimestamp`
-pkg/util/log/redact.go | `reflect.TypeOf(true)`
+pkg/util/hlc/timestamp.go | `Timestamp`
+vendor/github.com/cockroachdb/pebble/internal/humanize/humanize.go | `FormattedString`
```
Comment (irfansharif) [inline docs/generated/redact_safe.md:37]: Shouldn't this be sorted at the end?

Comment (knz) [reply docs/generated/redact_safe.md:37]: It is! It's at the end of the types that implement `SafeValue`.

The line afterwards is a different block, with the list of calls to `RegisterSafeType`.

Comment (irfansharif) [reply docs/generated/redact_safe.md:37]: Ah, LGTM.

**Review by irfansharif — APPROVED** [review]

**Comment by knz** [comment]
thanks!

bors r=irfansharif

**Pull request merged**